config/confighttp/confighttp.go

```diff
@@ -74,7 +75,7 @@ type ClientConfig struct {
 	Headers map[string]configopaque.String `mapstructure:"headers,omitempty"`
 
 	// Auth configuration for outgoing HTTP calls.
-	Auth *configauth.Config `mapstructure:"auth,omitempty"`
+	Auth configoptional.Optional[configauth.Config] `mapstructure:"auth,omitempty"`
```

Comment:
is zero-value of Optional a None value?

Reply:
also, I am curious how `omitempty` works for Optional. It seems like the output formatter, when do we use it?

Reply:
> is zero-value of Optional a None value?

Yep

> also, I am curious how omitempty works for Optional. It seems like the output formatter, when do we use it?

`omitempty` will omit `None` values (since it omits any zero values). It's used by `confmap.Marshal` which is used for things like OpAMP (to report the effective configuration to a backend)